shm/Game.cpp

```diff
@@ -0,0 +1,75 @@
+#include "Game.hpp"
+
+#include <iostream>
+
+constexpr size_t distancePerDay{2};
+
+Game::Game(size_t money, size_t days, size_t final_goal)
+    : money_(money), days_(days), final_goal_(final_goal) {}
+
+bool Game::checkWinCondition() const {
+    return (player_->getMoney() >= final_goal_);
+}
+
+bool Game::checkLoseCondition() const {
+    return (player_->getMoney() == 0);
+}
+
+void Game::Travel() {
+    std::cout << *map_;
+    auto destination = map_->getIsland(getTravelLocation());
+    if (!destination) {
+        return;
+    }
+
+    auto distance = map_->getDistanceToIsland(destination);
+    map_->setCurrentPosition(destination);
+    advanceTimeTraveling(distance);
+}
+
+Coordinates Game::getTravelLocation() {
+    std::cout << "Type position X of an Island to travel to: ";
+    size_t X{};
+    std::cin >> X;
+    std::cout << "\nType position Y of an Island to travel to: ";
+    size_t Y{};
+    std::cin >> Y;
+    std::cout << "\n";
+
+    return Coordinates(X, Y);
+}
```

Comment:
Still, we have to remember to somehow and somewhere validate coords given by user. 

Btw. we can display our islands on map by numbers or names 1, 2,... 10 or A, B, C, etc. Then user would have to provide only one character and we can easily validate if it's between `1` and `amountOfIslands`.

We should also give some kind of feedback about chosen island if it's incorrect.